File: Range_Sum_of_BST/solution.best.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
  public int rangeSumBST(TreeNode root, int L, int R) {
    if (root == null) return 0;
    
    int sum = (L <= root.val && root.val <= R) ? root.val : 0;
    
    if (root.val >= R) {
      return sum + this.rangeSumBST(root.left, L, R);
    } else if (root.val <= L) {
      return sum + this.rangeSumBST(root.right, L, R);
    } else {
      return sum + this.rangeSumBST(root.left, L, R) + this.rangeSumBST(root.right, L, R);
    }
  }
}

File: Range_Sum_of_BST/solution.best.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
  def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
    if root == None: return 0
    
    ans = root.val if L <= root.val and root.val <= R else 0
    
    if root.val >= R:
      return ans + self.rangeSumBST(root.left, L, R)
    elif root.val <= L:
      return ans + self.rangeSumBST(root.right, L, R)
    else:
      return ans + self.rangeSumBST(root.left, L, R) + self.rangeSumBST(root.right, L, R)
    
